### PR TITLE
Read `DD_GIT_REPOSITORY_URL` and `DD_GIT_COMMIT_SHA` env vars

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Make the agent read the containers `DD_GIT_REPOSITORY_URL` and `DD_GIT_COMMIT_SHA` environment variables to populate the `git.repository_url` and `git.commit.sha` tags.

### Motivation

Those `git.repository_url` and `git.commit.sha` tags are currently filled from the values of the `org.opencontainers.image.source` and `org.opencontainers.image.revision` docker labels.
But those labels are not used by the APM tracer libraries.
In order to ease the setup for the customers, it would be simpler to have one way to configure the repository URL and the commit sha that would work for both the agent and the APM tracer libraries.

### Possible Drawbacks / Trade-offs

This works only for the env vars set at the container level. If the env var is set by a script running inside the container, it won’t be seen by the agent.

### Describe how to test/QA your changes

Start a container with the repository URL and commit sha set as docker labels `DD_GIT_REPOSITORY_URL` and `DD_GIT_COMMIT_SHA`.
Check that all the data coming from this container are properly tagged with `git.repository_url` and `git.commit.sha`.

### CODE REVIEW

File: comp/core/tagger/taggerimpl/collectors/workloadmeta_extract.go [hunk 1]
```diff
@@ -72,6 +72,9 @@ var (
 	}
 
 	lowCardOrchestratorEnvKeys = map[string]string{
+		"DD_GIT_COMMIT_SHA":     tags.GitCommitSha,
```
Comment (vboulineau) [inline comp/core/tagger/taggerimpl/collectors/workloadmeta_extract.go:75]: Any test for that?

**Review by clamoriniere — COMMENTED** [review]
Hi @L3n41c

I'm not sure about the feature.

I understand that the APM librairies is not able to get the dockerfile label, however I feel like the APM library don't need to know about these labels or envvars, because if Origin detection works properly the `git.*` tags will be added by the trace-agent when it will call the tagger with the container_id? 

did I missed something? 


also if the idea is to provide to the user, to provide the git sha/repository outside the dockerfile. I think we will need also to add the support with Pod annotations to have feature parity with other Unified service taggins (env,service,version) WDYT?

File: .gitlab/container_build/docker_linux.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@
...
+    - docker buildx build --no-cache --push --pull --platform linux/$ARCH --build-arg CIBUILD=true --build-arg GENERAL_ARTIFACTS_CACHE_BUCKET_URL=${GENERAL_ARTIFACTS_CACHE_BUCKET_URL} $BUILD_ARG --build-arg DD_GIT_REPOSITORY_URL=https://github.com/DataDog/datadog-agent --build-arg DD_GIT_COMMIT_SHA=${CI_COMMIT_SHA} --file $BUILD_CONTEXT/Dockerfile --tag ${TARGET_TAG} --label "org.opencontainers.image.created=$(date --rfc-3339=seconds)" --label "org.opencontainers.image.authors=Datadog <package@datadoghq.com>" --label "org.opencontainers.image.source=https://github.com/DataDog/datadog-agent" --label "org.opencontainers.image.version=$(inv agent.version)" --label "org.opencontainers.image.revision=${CI_COMMIT_SHA}" --label "org.opencontainers.image.vendor=Datadog, Inc." $BUILD_CONTEXT
```
Comment (clamoriniere) [inline .gitlab/container_build/docker_linux.yml:21]: no strong opinion but maybe split the feature implementation from the change we do to use the feature for the datadog-agent images

File: releasenotes/notes/source-code-integration-env-var-c94278ccaa974c6e.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    The agent now populates the `git.repository_url` and `git.commit.sha` tags from the values of the `DD_GIT_REPOSITORY_URL` and `DD_GIT_COMMIT_SHA` container environment variables.
```
Comment (urseberry) [inline releasenotes/notes/source-code-integration-env-var-c94278ccaa974c6e.yaml:11]:
```suggestion
    The Agent now populates the `git.repository_url` and `git.commit.sha` tags from the values of the `DD_GIT_REPOSITORY_URL` and `DD_GIT_COMMIT_SHA` container environment variables.
```

**Review by urseberry — APPROVED** [review]

**Comment by bric3** [comment]
> I'm not sure about the feature.

@clamoriniere Hi, so the goal is to be able to tag events (like logs) with the repo. This would allow to search logs by repository which would be useful for IDE plug-ins where association services with code is not easily doable.

Also this would unify (and simplify) the configuration: Source Code Integration already needs this environment variable and agent was using OCI labels, now consumers will only have to set these environment variables.

File: comp/core/tagger/taggerimpl/collectors/workloadmeta_extract.go (hunk 1 above)
Comment (L3n41c) [reply comp/core/tagger/taggerimpl/collectors/workloadmeta_extract.go:75]: Sure!
Here is the corresponding test: https://github.com/DataDog/datadog-agent/pull/27279/files#diff-0ef59cd615f8a571da040730171ca3cd526cd7cad473989015888c5748bf0270R1480

**Review by clamoriniere — APPROVED** [review]
This PR will allow to improve the user experience in a subset of environment. 
however we will need to work on a more long term solution to have the same tagging parting across all the product.

**Review by clamoriniere — APPROVED** [review]
This PR will allow to improve the user experience in a subset of environment. 
however we will need to work on a more long term solution to have the same tagging parting across all the product.

**Review by Pythyu — APPROVED** [review]

**Pull request merged**